packages/yoastseo/src/languageProcessing/languages/en/helpers/internal/getAdjectiveStem.js

```diff
@@ -1,28 +1,68 @@
 // "use strict";
 import { languageProcessing } from "yoastseo";
+import { countSyllablesInWord } from "../../../../helpers/syllables/countSyllables";
+import syllables from "../../config/syllables.json";
 const { buildFormRule, createRulesFromArrays } = languageProcessing;
 
 /**
  * Constructs a function that checks if the input word can be a specific adjectival form.
  *
  * @param {string}      endsWith            How the form ends.
  * @param {int}         minimumWordLength   How long the word should be to classify for this form.
- * @param {string[]}    exceptions          The list of words with that ending (endsWith) which are not this form.
+ * @param {string[]}    nonStemmingExceptions  The list of words with that ending (endsWith) which are not this form.
+ * @param {string[]}    forcedStemmingExceptions The list of words with that ending (endsWith) which are not caught by the regular rules.
+ * @param {Function}    extraCheck          An extra check to determine if the word is this form.
  *
  * @returns {Function} A function that checks if the input word can be a specific adjectival form.
  */
-const constructCanBeFunction = function( endsWith, minimumWordLength, exceptions ) {
-	return word => {
+const constructCanBeFunction = function( endsWith, minimumWordLength, nonStemmingExceptions = [], forcedStemmingExceptions = [], extraCheck = null ) {
+	return ( word ) => {
 		const wordLength = word.length;
+
 		if ( wordLength < minimumWordLength ) {
 			return false;
 		}
-
-		const doesEndWith = word.substring( wordLength - endsWith.length, wordLength ) === endsWith;
-		return doesEndWith && ! exceptions.includes( word );
+		if ( extraCheck ) {
+			return extraCheck( word, endsWith, nonStemmingExceptions, forcedStemmingExceptions );
+		}
+		return word.endsWith( endsWith ) && ! nonStemmingExceptions.includes( word );
```

Comment:
I don't like this `null` pattern for the function parameter 😅 Can we have the second as the default function? So something like:

```javascript
const endsWithAndNoException = (word, endsWith, nonStemmingExceptions) => word.endsWith( endsWith ) && ! nonStemmingExceptions.includes( word );

const constructCanBeFunction = ( endsWith, minimumWordLength, nonStemmingExceptions = [], forcedStemmingExceptions = [], checkFunction = endsWithAndNoException ) => {

...

return checkFunction( word, endsWith, nonStemmingExceptions, forcedStemmingExceptions );
```